### PR TITLE
[SYCL] Ensure correct sycl::errc for SYCL2020

### PR DESCRIPTION
**Pull request by cperkinsintel**
The SYCL2020 specification has quite a few errc values it expects thrown exceptions to have. Most of the existing ones are fine, but there are several cases that need updating.  After this there are some other changes that will need to be done for errc conformance, but those changes are slightly more involved and I did not undertake them here. I'll open separate PRs for those.  

This PR may seem to touch a lot of files, but they are each small independent accesses. If it's easier to grok, just go through the individual commits.   

The tests were updated as well. https://github.com/intel/llvm-test-suite/pull/857 has the matching changes.

### CODE REVIEW

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/857

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/857

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/857

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/857

File: sycl/include/CL/sycl/accessor.hpp [hunk 1]
```diff
@@ -1443,7 +1443,14 @@ class __SYCL_SPECIAL_CLASS accessor :
                          detail::getSyclObjImpl(BufferRef).get(), Dimensions,
                          sizeof(DataT), BufferRef.OffsetInBytes,
                          BufferRef.IsSubBuffer) {
-    checkDeviceAccessorBufferSize(BufferRef.size());
+    preScreenAccessor(BufferRef.size(), PropertyList);
+    if (BufferRef.isOutOfBounds(AccessOffset, AccessRange,
+                                BufferRef.get_range()))
+      throw cl::sycl::invalid_object_error(
```
Comment (alexbatashev) [inline sycl/include/CL/sycl/accessor.hpp:1449]:
```suggestion
      throw sycl::invalid_object_error(
```

**Review by alexbatashev — APPROVED** [review]
LGTM. Some stylistic nits.

File: sycl/include/CL/sycl/accessor.hpp (hunk 1 above)
Comment (cperkinsintel) [reply sycl/include/CL/sycl/accessor.hpp]: done

File: sycl/include/CL/sycl/backend.hpp [hunk 2]
```diff
@@ -69,7 +69,8 @@ auto get_native(const SyclObjectT &Obj)
     -> backend_return_t<BackendName, SyclObjectT> {
   // TODO use SYCL 2020 exception when implemented
   if (Obj.get_backend() != BackendName) {
-    throw runtime_error("Backends mismatch", PI_INVALID_OPERATION);
+    throw runtime_error(errc::backend_mismatch, "Backends mismatch",
```
[the remark this answers is not in the document]
Comment (cperkinsintel) [reply sycl/include/CL/sycl/backend.hpp]: done

**Review by alexbatashev — APPROVED** [review]

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/857

**Comment by cperkinsintel** [comment]
ping to someone who can merge.  @romanovvlad  or @bader  or others.  This and its brother-in-test https://github.com/intel/llvm-test-suite/pull/857 are ready and approved.

**Pull request merged**